solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -376,10 +373,8 @@ public NamedList<Object> getLatestVersion() throws IOException {
     params.set(CommonParams.WT, JAVABIN);
     params.set(CommonParams.QT, ReplicationHandler.PATH);
     QueryRequest req = new QueryRequest(params);
-    req.setBasePath(leaderBaseUrl);
-    // TODO modify to use shardhandler
     try {
-      return solrClient.request(req, leaderCoreName);
+      return solrClient.requestWithBaseUrl(leaderCoreUrl, req).getResponse();
```

Comment:
[-1] Starting with [SOLR-17066](https://issues.apache.org/jira/browse/SOLR-17066) we're trying to move away from the use of core-URLs in SolrJ.  So we'll want to rework the usage that crept back in here.

We could address this by:

- adding a 'collection' arg to the lambda-less `requestWithBaseUrl`, or...
- creating a `requestWithBaseUrl` override with the 'collection' arg, or...
- using the existing lambda method:

    ```
        return solrClient.requestWithBaseUrl(leaderBaseUrl, (c) -> {
        return c.request(leaderCoreName, req);
        });
    ```

Any preference among those options @dsmiley, since you tweaked this bit recently?  If I don't hear from you in a few days I'll add a collection arg to the existing method, but I'm happy to go either route if you chime in.

Reply:
The whole collection ambiguity aspect duplicating our SolrJ methods is really a shame, but it's a problem for another JIRA.  In the mean time, doubling up requestWithBaseUrl (lambda-less) is my preference.

Also note: `org.apache.solr.client.solrj.SolrRequest#getCollection` but only used for CloudSolrClient.  What a mess IMO.

Reply:
or don't even double it; just have the collection variant and caller passes null if to node level.

Reply:
> The whole collection ambiguity aspect duplicating our SolrJ methods is really a shame

Agreed - I think we're in a better position to do something about that for 10.0 now that core-urls are dead (and `SolrRequest.setBasePath` soon will be as well).  But another JIRA for sure.

For now I'll go with the single method, `requestWithBaseUrl, baseUrl, collection, SolrRequest)`, and note in the Javadocs that collection may be null for "node" requests or "admin" requests or whatever we're calling them.

Reply:
(Done)